src/Syntax/Traversable.hs

```diff
@@ -0,0 +1,64 @@
+{-# LANGUAGE DefaultSignatures, EmptyCase, FlexibleContexts, FlexibleInstances, LambdaCase, MultiParamTypeClasses, RankNTypes, QuantifiedConstraints, TypeOperators #-}
+module Syntax.Traversable
+( HTraversable(..)
+, GHTraversable(..)
+) where
+
+import GHC.Generics
+import Syntax.Foldable
+import Syntax.Functor
+import qualified Syntax.Sum as Sum
+
+class ( HFoldable sig
+      , HFunctor sig
+      , forall g . Functor  g    => Functor     (sig g)
+      , forall g . Traversable g => Traversable (sig g)
+      )
+   => HTraversable sig where
+  htraverse
+    :: (Monad f, Traversable g)
+    => (forall a . g a -> f (h a))
+    -> (sig g a -> f (sig h a))
```

Comment:
I’m more confident in this definition than in `HFoldable`.

Note that unlike `HFunctor` (defined in `fused-effects`), we have quantified superclass constraints which greatly improve the convenience of using the class with nested syntax.